From 5bc2a6405d495ea6c6296933ed7f25f2be7a385d Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Fri, 12 Jul 2024 11:36:09 +0200
Subject: [PATCH 1159/1695] drivers: stm32mp_pmic2: leverage qsort_int() in
 PMIC driver

Use recently added qsort_int() helper function.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 67474724a73b6e7804a370a96aed661d8f8240f8 drivers: stm32mp_pmic2: leverage qsort_int() in PMIC driver]
Change-Id: Ic48bda57393d65b4ea4aac4a979e4f2155e55975
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/393544
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/drivers/stm32mp_pmic2.c | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/core/drivers/stm32mp_pmic2.c b/core/drivers/stm32mp_pmic2.c
index 2a2dd29a8..e7426d34f 100644
--- a/core/drivers/stm32mp_pmic2.c
+++ b/core/drivers/stm32mp_pmic2.c
@@ -234,21 +234,13 @@ out:
 	return res;
 }
 
-static int cmp_int_value(const void *a, const void *b)
-{
-	const int *ia = a;
-	const int *ib = b;
-
-	return CMP_TRILEAN(*ia, *ib);
-}
-
 static size_t refine_levels_array(size_t count, int *levels_uv,
 				  int min_uv, int max_uv)
 {
 	size_t n = 0;
 	size_t m = 0;
 
-	qsort(levels_uv, count, sizeof(*levels_uv), cmp_int_value);
+	qsort_int(levels_uv, count);
 
 	/* Remove duplicates and return optimized count */
 	for (n = 1; n < count; n++) {
-- 
2.47.2

